JavaShared/src/main/java/org/openzen/zenscript/javashared/JavaContext.java

```diff
@@ -644,4 +644,8 @@ public String getMethodSignatureConstructor(MethodSymbol method) {
 
 		return getMethodSignature(javaHeader, true, true);
 	}
+
+	public boolean isGenericOrContainsGenericParameters(TypeID typeID) {
```

Comment:
Something probably got lost in the rebase-foo:
This method is unused (and the JavaTypeCheckIfGenericVisitor is only used here so it's transitively-unused 😛 )

This was the original commit: https://github.com/ZenCodeLang/ZenCode/commit/4bbd17d27ceee47f17e09cf7549666ba38030efd

Might be covered by `JavaContext#isGenericReturn` ?
...or that method could instead use this visitor?

```java
	public boolean isGenericReturn(TypeID type) {
		return type.isGeneric() || type.asArray().map(array -> isGenericReturn(array.elementType)).orElse(false);
	}
```